osdep/terminal-unix.c

```diff
@@ -358,8 +358,10 @@ static void stop_sighandler(int signum)
 {
     int saved_errno = errno;
     (void)write(stop_cont_pipe[1], &(char){PIPE_STOP}, 1);
-    (void)write(STDERR_FILENO, TERM_ESC_RESTORE_CURSOR,
-                sizeof(TERM_ESC_RESTORE_CURSOR) - 1);
+    if (isatty(STDERR_FILENO)) {
```

Comment:
`isatty` isn't listed as [signal-safe](https://man7.org/linux/man-pages/man7/signal-safety.7.html).

Also, failed to notice this before but I don't think it should be writing to stderr like this anyways. It should probably use `tty_out` (??)

Reply:
status line is always printed to stderr and it has been hardcoded for years like that in few places.

> isatty isn't listed as signal-safe.

fair point, but I don't see any other way to ensure we don't print garbage if stderr is redirected (not tty)

Reply:
> I don't see any other way to ensure we don't print garbage if stderr is redirected

The processing can be moved out of the handler. See #13238.

> status line is always printed to stderr and it has been hardcoded for years like that in few places.

Hmm, I see. Should I keep it hardcoded to stderr instead of `tty_out` then?

Reply:
> The processing can be moved out of the handler.

Good.

> Hmm, I see. Should I keep it hardcoded to stderr instead of tty_out then?

I think so, we do hide cursor on stderr only if it is a tty. So for symmetry we should restore it similarly.